internal/proxy/proxy.go

```diff
@@ -151,10 +160,13 @@ var (
 	// 'projects/<PROJECT>/locations/<REGION>/clusters/<CLUSTER>/instances/<INSTANCE>'
 	// Additionally, we have to support legacy "domain-scoped" projects (e.g. "google.com:PROJECT")
 	instURIRegex = regexp.MustCompile("projects/([^:]+(:[^:]+)?)/locations/([^:]+)/clusters/([^:]+)/instances/([^:]+)")
```

Comment:
Optional Nit:  If you wanted to be really clever, use a non-capturing group `(?:   )`  for the sub-domain in PROJECT group

```
instURIRegex = regexp.MustCompile(
    "projects/([^:]+(?::[^:]+)?)/locations/([^:]+)/clusters/([^:]+)/instances/([^:]+)")
```
Then the so that your code down below goes 1,2,3,4:

```
project := string(m[1])
region := string(m[2])
cluster := string(m[3])
name := string(m[4])
```

Reply:
Fancy -- I'm working on making this code exported from the connector and will revisit this change with unit tests when I get there.

Reply:
See https://github.com/GoogleCloudPlatform/alloydb-auth-proxy/pull/137.